csrc/ops/indexing.cpp

```diff
@@ -74,6 +74,10 @@ TensorView* indexSelect(
         "index_tv must be a 1d or scalar tensor.");
     index_tv =
         ops::maybeBroadcastIndexTv(index_tv->as<TensorView>(), dim, n_dims);
+  } else {
+    // We do an explicit set here, which would allow vectorized load on index_tv when applicable.
+    index_tv =
+        set(index_tv->as<TensorView>());
```

Comment:
without this copy, we wouldn't have vectorized load on index_tv. Hence would fail the added test IndexSelectVectorizationIndexTensorNoBroadcast